atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/VersionedEventStore.java

```diff
@@ -48,11 +50,25 @@
                 .orElseGet(ImmutableList::of);
     }
 
-    LockWatchEvents retentionEvents() {
-        int numToRetention = Math.max(0, eventMap.size() - maxEvents);
+    LockWatchEvents retentionEvents(Optional<Long> earliestSequenceToKeep) {
```

Comment:
This overall makes sense, but as you pointed out, it can cause the size of this to grow almost unboundedly. We should probably do two things:
a) come up with some limit on number of events where we retention the old events anyway
b) have a way for transactions to declare they are / are not using LW, so if we have slow transactions that do not actually use them, we can still retention away